### DISCUSSION TITLE
How can i enable Swagger UI using docker image

### DISCUSSION THREAD

**Discussion by libinmath3w** [Keyfactor/ejbca-ce#177] [Q&A]
Hi,
  I tried to enable Swagger UI  in my EJBCA server running via docker image. However, i didn't found any way to enable it. How can i enable  Swagger UI in a docker image. ?

Thank  you

**Comment by mike-agrenius-kushner** [comment]
Hi, 

The swagger UI is not deployed on the container (because the flag ejbca.productionmode=true) is set there, since we consider it a security risk and don't deploy it by default. 

If you can make do you can use the REST API reference (https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/protocols/ejbca-rest-interface#EJBCARESTInterface-rest_api_reference) or download EJBCA as source and build with ejbca.productionmode=false in ejbca.properties

Cheers,
Mike

**Reply by timoggit** [reply]
Hi,
i have the same problem - i cannot see / use the swagger UI. 
we are using the software appliances and the docker images.
and we do not want or can deploy the sources by ourself...

is there any other possibility to get swagger or the REST API working?

cheers
Timo

**Reply by primetomas** [reply]
The REST API works without swagger, you can use curl or anything else. You need to start the container with "classic workflow", client certificate authentication to use the REST API.
You can find example here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/protocols/ejbca-rest-interface
And a java CLI client here: https://github.com/Keyfactor/ejbca-easy-rest-client

**Comment by libinmath3w** [comment]
thanks @mike-agrenius-kushner . i want to generate TLS and other certificates using api. However, the documentation  mentioned below lines which are not showing my ejbca server.

> Each EJBCA release is bundled with an auto-generated REST API reference available on the URL https://localhost:8442/ejbca/doc/rest/ 
if deployed.

if i visit  example.com/ejbca/doc/rest/  then i get 404 error page. i have already enabled api in protocol section.   how can i fix this issues.

![Screenshot (412)](https://user-images.githubusercontent.com/24878821/204537769-f5d09dbc-e7f6-485c-a276-44bb503cf6bb.png)

**Comment by primetomas** [comment]
See #661 

You can download swagger.json from GitHub: https://github.com/Keyfactor/ejbca-ce/blob/main/modules/ejbca-rest-api/resources/swagger.json

Then you can run swagger locally by for exampel:
docker run -p 80:8080 -e SWAGGER_JSON=/foo/swagger.json -v /tmp:/foo swaggerapi/swagger-ui

That gives you Swagger API.
